jgrapht-core/src/main/java/org/jgrapht/alg/planar/ElementList.java

```diff
@@ -0,0 +1,438 @@
+     * This method throws {@link NoSuchElementException} in the case {@code element} doesn't
+     * belong to this list
+     *
+     * @param element an element to start an iteration from
+     * @return an iterator over this list, which starts from the {@code element}
+     */
+    public Iterator<E> reverseIteratorFrom(E element) {
+        ListNode<E> start = getNode(element);
+        if (start == null) {
+            throw new NoSuchElementException();
+        }
+        return new ListIterator<>(start, true);
+    }
+
+    /**
+     * Returns true if this list is empty, false otherwise
+     *
+     * @return true if this list is empty, false otherwise
+     */
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    /**
+     * Returns the number of elements in this list
+     *
+     * @return the number of elements in this list
+     */
+    public int getSize() {
+        return size;
+    }
+
+    /**
+     * Appends the specified value to the end of this list. This method
+     * returns the list node the {@code value} is contained in
+     *
+     * @param value the value to append to this list
+     * @return the list node, which contains the {@code value}
+     */
+    public ListNode<E> add(E value) {
+        return append(value);
+    }
+
+    /**
+     * Inverts the list. For instance, calling this method on the list $(a,b,c,\dots,x,y,z)$
+     * will result in the list $(z,y,x,\dots,c,b,a)$. This method does only pointer manipulation,
+     * meaning that all the list nodes allocated for the previously added elements are valid after
+     * this method finishes.
+     */
+    public void invert() {
+        if (size < 2) {
+            return;
+        }
+        ListNode<E> newHead = head.prev;
+        ListNode<E> current = head;
+        do {
+            ListNode<E> t = current.next;
+            current.next = current.prev;
+            current.prev = t;
+            current = current.prev;
+        } while (current != head);
+        head = newHead;
+    }
+
+    /**
+     * Appends the {@code list} to the end of this list. All the elements from {@code list}
+     * are transferred to this list, i.e. the {@code list} is empty after calling this method.
+     *
+     * @param list the list to append
+     */
+    public void append(ElementList<E> list) {
+        if (!list.isEmpty()) {
+            if (isEmpty()) {
+                head = list.head;
+            } else {
+                linkBefore(head, list);
+            }
+            size += list.size;
+            list.size = 0;
+            list.head = null;
+        }
+
+    }
+
+    /**
+     * Prepends the {@code list} to the beginning of this list. All the elements from {@code list}
+     * are transferred to this list, i.e. the {@code list} is empty after calling this method.
+     *
+     * @param list the list to prepend
+     */
+    public void prepend(ElementList<E> list) {
```

Comment:
I think it would be better to use similar names like the `ArrayDeque`. This would make it easier for other Java users.   For example "prepend" -> "addFirst" and "append" -> "addLast". Same comment applies in other names also.